src/Sarif/Core/Run.cs

```diff
@@ -32,13 +136,13 @@ public bool ShouldSerializeColumnKind()
             return true;
         }
 
-        public bool ShouldSerializeFiles() { return this.Files != null && this.Files.Values.Any(); }
+        public bool ShouldSerializeFiles() { return this.Files.Any(); }
```

Comment:
>. [](start = 62, length = 1)

Here and on Line 145, you're assuming that these array-valued properties will be non-null (that is, that they will always be present, even if empty). That means that we'll unnecessarily serialize empty arrays, right? Shouldn't we let them be null, and the use `this.TheArrayValuedProperty?.Any()`?

Reply:
hm, my assumption may be wrong on looking at relevant source. will test and update appropriately. handling a null 'this' pointer is a cute trick but non-obvious is you're not tracking that the relevant method is an extension method. so it may be verboten in CLR API as breaking principle of least surprise. will follow up.

---
In reply to: [244604442](https://github.com/Microsoft/sarif-sdk/pull/1191#discussion_r244604442) [](ancestors = 244604442,244521595)